### DISCUSSION TITLE
What access permission is needed to create a Pull Request to public repositories?

### DISCUSSION THREAD

**Discussion by ryosuketc** [community/community#22665] [New to GitHub]
<p>What permission do we need to send a PR to public repositories which we have only read access? Only read or need write as well?</p>
<p>These two statements look conflicting to me.</p>
<p>Creating a pull request - GitHub Docs<br>
<a href="https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request</a></p>
<blockquote>
<p>Anyone with read permissions to a repository can create a pull request</p>
</blockquote>
<blockquote>
<p><strong>Note</strong> : To open a pull request in a public repository, you must have write access to the head or the source branch</p>
</blockquote>

**Comment by thinkverse** [comment, marked as answer]
<p>Hello. :wave:</p>
<p>That means you need to have write access to open a pull request and change the code <strong>directly</strong> in that public repository, which I something I don’t think GitHub offers for security reasons.</p>
<p>To contribute to a public repository you will have to fork that repository, creating your version which you will have write access to, write to that version, and then send a pull request from that version back to the original repository.</p>
<p>That’s how you usually go about it, but look out for CONTRIBUTING.md, LICENSE, and, or, CODE_OF_CONDUCT.md files, if a repository has those you should read them first before submitting any pull request to make sure you’ve followed all the necessary steps they require for a pull request, and you should read the README.md too but that’s in the name so. :slightly_smiling_face:</p>
<p>Do keep in mind that if a repository doesn’t have a LICENSE, even if it’s visible and public, it’s not considered open-source since it doesn’t have a LICENSE, in that case, I would suggest opening an issue or contacting the owner directly and asking for permission to contribute or asking them for a license.</p>
<p>If you have a public repository and want to give write access to a contributor to edit code directly on the repository you can invite them under <code>settings &gt; manage access</code>.</p>
<p>Hope that helped. :grinning:</p>

**Comment by ryosuketc** [comment]
<p>Thank you thinkverse. It’s very clear now.<br>
I was wondering how a PR to Read-only repos works. When I do something like this:</p>
<pre><code class="lang-auto">git clone hogehoge  # a repository which I only have Read access
touch test.txt
git add .
git commit -m 'test'
gh pr create  # make a PR
</code></pre>
<p>It turned out that a fork is automatically created and any of my local branches are set up to track the fork, instead of the original repo.<br>
Hence, I think <strong>a direct PR to Read-only repos is simply impossible</strong> (i.e. PR is done through a fork). What’s possible is:</p>
<ul>
<li>Read access + PR through a fork</li>
<li>Write access + direct PR</li>
</ul>
<p>I also appreciate the detailed explanation on the documentations. It’s of great help :slight_smile:</p>

**Comment by thinkverse** [comment]
<aside class="quote no-group" data-username="ryosuketc" data-post="4" data-topic="129825">
<div class="title">
<div class="quote-controls"></div>
 ryosuketc:</div>
<blockquote>
<p>It turned out that a fork is automatically created and any of my local branches are set up to track the fork, instead of the original repo.</p>
</blockquote>
</aside>
<p>Correct, that is in fact how the <code>git clone</code> command works, this excerpt is taken from the <a href="https://git-scm.com/docs/git-clone#_description" rel="nofollow noopener">documentation</a> on <code>git clone</code>.</p>
<blockquote>
<p>and creates and checks out an initial branch that is forked from the cloned repository’s currently active branch.</p>
</blockquote>
<p>I’m glad I could help, good luck with your journey. :+1:</p>

**Comment by ryosuketc** [comment]
<p>Thank you! It’ clear now.</p>

**Comment by thinkverse** [comment]
<p>Glad I could help, if one post helped you solved it, please mark it as the solution, it will help people looking around for the same issue in the future. :slightly_smiling_face:</p>
